standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/cache/SharedCache.java

```diff
@@ -715,154 +747,105 @@ public void removeConstraint(String name) {
       }
     }
 
-    public void refreshPrimaryKeys(List<SQLPrimaryKey> keys) {
-      Map<String, SQLPrimaryKey> newKeys = new ConcurrentHashMap<>();
+    public void refreshAllTableConstraints(SQLAllTableConstraints constraints) {
       try {
         tableLock.writeLock().lock();
-        int size = 0;
-        for (SQLPrimaryKey key : keys) {
-          if (compareAndSetMemberCacheUpdated(MemberName.PRIMARY_KEY_CACHE, true, false)) {
-            LOG.debug("Skipping primary key cache update for table: " + getTable().getTableName()
-                    + "; the primary keys are already refreshed.");
-            return;
-          }
-          String pkName = StringUtils.normalizeIdentifier(key.getPk_name());
-          key.setPk_name(pkName);
-          newKeys.put(pkName, key);
-          size += getObjectSize(SQLPrimaryKey.class, key);
-        }
-        primaryKeyCache = newKeys;
-        updateMemberSize(MemberName.PRIMARY_KEY_CACHE, size, SizeMode.Snapshot);
-        LOG.debug("Primary keys refresh in cache was successful for {}.{}.{}",
-            this.getTable().getCatName(), this.getTable().getDbName(), this.getTable().getTableName());
+        this.isConstraintsValid =
+            refreshConstraint(constraints.getPrimaryKeys(), MemberName.PRIMARY_KEY_CACHE) && refreshConstraint(
+                constraints.getForeignKeys(), MemberName.FOREIGN_KEY_CACHE) && refreshConstraint(
+                constraints.getUniqueConstraints(), MemberName.UNIQUE_CONSTRAINT_CACHE) && refreshConstraint(
+                constraints.getDefaultConstraints(), MemberName.DEFAULT_CONSTRAINT_CACHE) && refreshConstraint(
+                constraints.getNotNullConstraints(), MemberName.NOTNULL_CONSTRAINT_CACHE) && refreshConstraint(
+                constraints.getCheckConstraints(), MemberName.CHECK_CONSTRAINT_CACHE);
       } finally {
         tableLock.writeLock().unlock();
       }
     }
 
-    public void refreshForeignKeys(List<SQLForeignKey> keys) {
-      Map<String, SQLForeignKey> newKeys = new ConcurrentHashMap<>();
-      try {
-        tableLock.writeLock().lock();
-        int size = 0;
-        for (SQLForeignKey key : keys) {
-          if (compareAndSetMemberCacheUpdated(MemberName.FOREIGN_KEY_CACHE, true, false)) {
-            LOG.debug("Skipping foreign key cache update for table: " + getTable().getTableName()
-                    + "; the foreign keys are already refreshed.");
-            return;
+    private boolean refreshConstraint(List constraints, MemberName mn) {
```

Comment:
This also always returns true and in tune make isConstraintsValid=true always.

Reply:
Yes